python/opaque/__init__.py

```diff
@@ -355,78 +355,89 @@ def UserAuth(ctx, auth, infos):
 #  to supply a long-term pubkey as a parameter, this might be one
 #  unique global key, or it might be a per-user key derived from a
 #  server secret.
-#int opaque_Create1kRegistrationResponse(const uint8_t M[crypto_core_ristretto255_BYTES], const uint8_t pkS[crypto_scalarmult_BYTES], uint8_t _sec[OPAQUE_REGISTER_SECRET_LEN], uint8_t _pub[OPAQUE_REGISTER_PUBLIC_LEN]) {
-def Create1kRegistrationResponse(alpha, pk):
-    if None in  (alpha, pk):
+#
+#  This function is called CreateRegistrationResponse in the rfc.
+#  The server receives M from the users invocation of its
+#  opaque_CreateRegistrationRequest() function, it outputs a value sec
+#  which needs to be protected until step 4 by the server. This
+#  function also outputs a value pub which needs to be passed to the
+#  user.
+#  @param [in] M - the blinded password as per the OPRF.
+#  @param [in] pkS - the servers long-term pubkey
+#  @param [out] sec - the private key and the OPRF secret of the server.
+#  @param [out] pub - the evaluated OPRF and pubkey of the server to
+#  be passed to the client into opaque_FinalizeRequest()
+#  @return the function returns 0 if everything is correct.
+#int opaque_Create1kRegistrationResponse(const uint8_t M[crypto_core_ristretto255_BYTES], const uint8_t pkS[crypto_scalarmult_BYTES], uint8_t sec[OPAQUE_REGISTER_SECRET_LEN], uint8_t pub[OPAQUE_REGISTER_PUBLIC_LEN]);
+def Create1kRegistrationResponse(M, pkS):
+    if None in  (M, pkS):
         raise ValueError("invalid parameter")
-    if len(alpha) != 32: raise ValueError("invalid alpha param")
-    if len(pk) != pk: raise ValueError("invalid pk param")
+    if len(M) != 32: raise ValueError("invalid M param")
+    if len(pkS) != pkS: raise ValueError("invalid pkS param")
 
     sec = ctypes.create_string_buffer(OPAQUE_REGISTER_SECRET_LEN)
     pub = ctypes.create_string_buffer(OPAQUE_REGISTER_PUBLIC_LEN)
-    __check(opaquelib.opaque_Create1kRegistrationResponse(alpha, pk, sec, pub))
+    __check(opaquelib.opaque_Create1kRegistrationResponse(M, pkS, sec, pub))
     return sec.raw, pub.raw
 
 #  Client finalizes registration by concluding the OPRF, generating
 #  its own keys and enveloping it all.
 #
 #  This function is called FinalizeRequest in the rfc.  This function
-#  is run by the user, taking as input the context ctx that was an
-#  output of the user running opaque_private_init_usr_start(), and the
-#  output pub from the server of opaque_private_init_srv_respond().
-#  The key parameter can be used as an extra contribution to the
-#  derivation of the rwd by means of being used as a key to the final
-#  hash, if not specified it uses the value specified by the rfc. The
+#  is run by the user, taking as input the context sec that was an
+#  output of the user running opaque_CreateRegistrationRequest(), and the
+#  output pub from the server of opaque_CreateRegistrationResponse().
+#  The info parameter can be used as an extra contribution to the
```

Comment:
is there still an info parameter?

Reply:
It is not. I removed it in https://github.com/stef/libopaque/pull/13/commits/eef80c760d14ec90b6af0b9506216955ccd0ab42 .